src/software/thunderscope/robot_diagnostics/chicker.py

```diff
@@ -46,8 +55,35 @@ def __init__(self, proto_unix_io):
         self.auto_kick_button = self.radio_buttons[1]
         self.auto_chip_button = self.radio_buttons[2]
 
-        vbox_layout.addWidget(self.radio_button_box)
+        # set buttons to be initially enabled
+        self.kick_chip_buttons_enable = True
+
+        # indicating that no auto button is selected by default
         self.no_auto_button.setChecked(True)
+        self.no_auto_selected = True
+
+        # adding onclick functions for buttons
+
+        # kick button and chip button connected to send_command_and_timeout with their respective commands
+        self.kick_button.clicked.connect(
+            lambda: self.send_command_and_timeout(ChickerCommandMode.KICK)
+        )
+        self.chip_button.clicked.connect(
+            lambda: self.send_command_and_timeout(ChickerCommandMode.CHIP)
+        )
+
+        # no auto button enables both buttons, while auto kick and auto chip disable both buttons
+        self.no_auto_button.toggled.connect(
+            lambda: self.set_should_enable_buttons(True)
+        )
```

Comment:
There's a bug now where if we select autokick/chip and then select no auto, the cached primitiveset with the old autokick/chip values will continue to be sent. So when the no auto button is selected we also need to send an empty primitiveset to reset the cache.

Reply:
makes sense, I tested my new changes and this looks fixed